doc/lsst.daf.butler/formatters.rst

```diff
@@ -0,0 +1,316 @@
+.. _daf_butler_storageclass_formatters_assemblers:
+
+##########################################
+Storage Classes, Assemblers and Formatters
+##########################################
+
+Formatters and assemblers provide the interface between Butler and the python types it is storing and retrieving.
+A Formatter is responsible for serializing a Python type to an external storage system and reading that serialized form back into Python.
+The serialization can be stored to a local file system or cloud storage or even a database.
+It is possible for a formatter to be globally configured to use particular parameters on write.
+On retrieval of datasets read parameters can be used that can, for example, return only a subset of the data.
+
+Assemblers are used to disassemble and reassemble composite datasets and can also be used to process read parameters that adjust how the retrieved dataset might be modified on get.
+
+Deciding which formatter or assembler to use is controlled by the storage class and corresponding dataset type.
+
```

Comment:
As mentioned above, a note here on how to get the default and current configs would be very helpful.  Looking at that output made the description of what all the parameters mean a lot clearer to me (on top of the examples, seeing the full set of options was very illuminating).